OpenRA.Game/Translation.cs

```diff
@@ -53,63 +63,63 @@ static IEnumerable<MessageContext> GetMessageContext(string language, string[] t
 				var stream = fileSystem.Open(path);
 				using (var reader = new StreamReader(stream))
 				{
-					var options = new MessageContextOptions { UseIsolating = false };
-					var messageContext = new MessageContext(language, options);
-					var errors = messageContext.AddMessages(reader);
-					foreach (var error in errors)
+					var parser = new LinguiniParser(reader);
+					var resource = parser.Parse();
+					foreach (var error in resource.Errors)
 						Log.Write("debug", error.ToString());
 
-					yield return messageContext;
+					bundle.AddResource(resource, out var errors);
```

Comment:
This isn't quite right either: the errors generated here are actually ignored, and if fixed in the line below generates a lot of false positives!

I expect that we want to stick with `AddResourceOverriding` here, but change paths definition above to something like
```csharp
// Always load english strings to provide a fallback for missing translations
var paths = translations.Where(t => t.EndsWith("en.ftl")).ToList();
foreach (var t in translations)
	if (t.EndsWith(language + ".ftl") && !paths.Contains(t))
		paths.Add(t);

foreach (var path in paths)
```
(note: not tested!).

This will load the backfalls first and override them with the proper strings. It also avoids parsing the same files twice for english.

Reply:
Can you provide me with your test case as I can't quite follow:

```fluent
## Server Orders

lobby-disconnected = { $player } hat die Lobby verlassen.
player-disconnected =
    { $team ->
        [0] { $player } hat die Verbindung beendet
       *[other] { $player } (Team { $team }) hat die Verbindung beendet
    }

master-server-connected = Verbindung mit Master Server hergestellt.
```

```yaml
Translations:
	common|languages/de.ftl
	common|languages/en.ftl
```

works as expected and I don't get any entry in `debug.log` for having incomplete translations.

Reply:
Change the line below this one to loop over `errors` instead of `resource.Errors`.

Reply:
Both variants worked in my tests.